datastore/core/key.py

```diff
@@ -190,12 +190,12 @@ def __len__(self):
     def __lt__(self, other):
         if isinstance(other, Key):
             return self._string < other._string
-        raise TypeError('other is not of type %s' % Key)
+        raise NotImplementedError('The second operand is not of type %s' % Key)
```

Comment:
This should `return NotImplemented` (a special built-in type), not directly raise `NotImplementedError`. The later being automatically raised by the Python interpreter when all possible combinations of the comparison (`A.__lt__(B)` and `B.__ge__(A)`) have been exhausted.

Reply:
Cool